modules/caddyhttp/reverseproxy/reverseproxy.go

```diff
@@ -792,10 +792,15 @@ func (lb LoadBalancing) tryAgain(ctx caddy.Context, start time.Time, proxyErr er
 	}
 
 	// otherwise, wait and try the next available host
+	timer := time.NewTimer(time.Duration(lb.TryInterval))
 	select {
-	case <-time.After(time.Duration(lb.TryInterval)):
+	case <-timer.C:
 		return true
 	case <-ctx.Done():
+		if !timer.Stop() {
+			// if the timer has been stopped then read from the channel
+			<-timer.C
```

Comment:
Do you think it might be a good idea to drain the timer in it's own goroutine if the `timer.Stop()` call doesn't stop the timer. For example something like- 
```
go func(){
	<-timer.C
}()
```
Also the comment on top of it is not very useful I think. Also, thanks TIL about this memory leak.

Reply:
I don't think so; see the docs: https://pkg.go.dev/time#Timer.Stop

Reply:
@mholt If the timer has already expired and `time.Stop()` wouldn't return true and `<-timer.C` would ensure a value *atleast* after the duration, but it can be very long too. Draining this in a goroutine would ensure the normal flow of the code from this point of view and draining elsewhere. Please let me know if I am missing something and sorry for being noob in that case.

Reply:
No, this only reads from the channel if the timer has already been stopped or expired. So the read will never be blocking in this case.